### PR TITLE
✨ Support likeWallet sign-in

### PR DESCRIPTION
**Pull request by nwingt**
This PR should be able to release before chain upgrade

### CODE REVIEW

File: util/User.js [hunk 1]
```diff
@@ -73,12 +73,15 @@ const User = {
     return data;
   },
 
-  async signCosmosLogin(cosmosWallet, signer, platform = 'cosmosWallet', loginMessage = LOGIN_MESSAGE) {
-    if (!cosmosWallet) return null;
+  async signCosmosLogin(inputWallet, signer, platform = 'cosmosWallet', loginMessage = LOGIN_MESSAGE) {
+    if (!inputWallet) return null;
+    if (platform !== 'cosmosWallet' && platform !== 'likeWallet') {
```
Comment (williamchong) [inline util/User.js:78]:
```suggestion
    if (`!['cosmosWallet ', 'likeWallet '].includes(platform)`) {
```

File: store/modules/getters/user.js [hunk 2]
```diff
@@ -12,7 +12,11 @@ export const getUserIsRegistered = state => !!state.user.user;
 
 export const getUserIsInBonusCoolDown = state => state.user.bonusCooldown;
 
-export const getUserHasWallet = state => !!(state.user.wallet || state.user.cosmosWallet);
+export const getUserHasWallet = state => !!(
+  state.user.wallet || state.user.cosmosWallet || state.user.likeWallet
```
Comment (williamchong) [inline store/modules/getters/user.js:16]: Can we remove `state.user.wallet`? This function would be more useful without eth wallet
We need to make sure metamask login flow does not use this function though

**Review by williamchong — CHANGES_REQUESTED** [review]

File: store/modules/getters/user.js (hunk 2 above)
Comment (nwingt) [reply store/modules/getters/user.js:16]: Can we just remove all ETH-related UI except for user login? Otherwise, that will make a lot of change

Comment (williamchong) [reply store/modules/getters/user.js:16]: hmm keep this change and leave for revamp?

File: pages/in/tx/history.vue [hunk 3]
```diff
@@ -7,8 +7,8 @@
 
           <transaction-history
             ref="txHistory"
-            :user="getUserInfo.user"
-            :address="getUserHasWallet"
+            :user="getUserId"
+            :address="getUserERC20WalletAddress"
```
Comment (williamchong) [inline pages/in/tx/history.vue:11]: The transaction API returns tx of cosmos too, so `getUserWalletAddress || getUserERC20WalletAddress`?
But this UI seems broken anyway

Comment (williamchong) [reply pages/in/tx/history.vue:11]: Or is this erc20 guard intentional?

Comment (nwingt) [reply pages/in/tx/history.vue:11]: User without ERC20 address will redirect to /in, I don't think users will come to this page anymore, there is no entry at all in like.co

**Comment by williamchong** [comment]
This PR will go to feature branch, I merge the API in master in a seperate commit i guess

**Pull request merged**